contrib/format-httpd/src/main/java/org/apache/drill/exec/store/httpd/HttpdLogBatchReader.java

```diff
@@ -40,36 +41,29 @@
 import java.io.InputStream;
 import java.io.InputStreamReader;
 
-public class HttpdLogBatchReader implements ManagedReader<FileSchemaNegotiator> {
+public class HttpdLogBatchReader implements ManagedReader {
 
   private static final Logger logger = LoggerFactory.getLogger(HttpdLogBatchReader.class);
   public static final String RAW_LINE_COL_NAME = "_raw";
   public static final String MATCHED_COL_NAME = "_matched";
   private final HttpdLogFormatConfig formatConfig;
-  private final int maxRecords;
-  private final EasySubScan scan;
-  private HttpdParser parser;
-  private FileSplit split;
+  private final HttpdParser parser;
+  private final FileDescrip file;
   private InputStream fsStream;
-  private RowSetLoader rowWriter;
+  private final RowSetLoader rowWriter;
   private BufferedReader reader;
   private int lineNumber;
-  private CustomErrorContext errorContext;
-  private ScalarWriter rawLineWriter;
-  private ScalarWriter matchedWriter;
+  private final CustomErrorContext errorContext;
+  private final ScalarWriter rawLineWriter;
+  private final ScalarWriter matchedWriter;
   private int errorCount;
 
-
-  public HttpdLogBatchReader(HttpdLogFormatConfig formatConfig, int maxRecords, EasySubScan scan) {
+  public HttpdLogBatchReader(HttpdLogFormatConfig formatConfig, EasySubScan scan, FileSchemaNegotiator negotiator) {
     this.formatConfig = formatConfig;
-    this.maxRecords = maxRecords;
-    this.scan = scan;
-  }
 
-  @Override
-  public boolean open(FileSchemaNegotiator negotiator) {
```

Comment:
There is a more fundamental consideration. Drill is distributed: we want to hold resources as short a time as possible. In the previous design, developers had to understand to not open files or obtain resources in the constructor. They had to remember to do that in `open()`. A side effect is that many variables where thus non-final.

Now, the readers follow the Java `Stream`, `Reader` and `Writer` convention: the constructor opens any external files, connections or other resources. `close()` releases them. The framework ensures that these operations span the shortest possible amount of time.

For background, in the pre-EVF (so-called "scan V1") code, it was common for people to open files in the constructor, but we'd create all the readers up front: so we'd have lots of open files, lots of unused buffers, etc. Now, the developer simply can't do things wrong: the constructor is called as late as possible, and it is the only place available to obtain resources. The scan calls `close()` as early as possible (and on an error) so resources can't leak or sit idle.

Again, sorry the scan is so complex. At least, with EVF, all the logic is in one place rather than copy-pasted across all readers (where it was often done wrongly) as in "scan V1."